packages/web-components/src/select/select.ts

```diff
@@ -0,0 +1,206 @@
+import { attr, FASTElement, observable } from '@microsoft/fast-element';
+import { ARIAGlobalStatesAndProperties } from '@microsoft/fast-foundation';
+import { SelectSize } from './select.options.js';
+
+/**
+ * The base class used for constructing a fluent-slider custom element
+ * @public
+ */
+export class Select extends FASTElement {
+  constructor() {
+    super();
+    this.handleChange = this.handleChange.bind(this);
+    this.handleInput = this.handleInput.bind(this);
+    Object.assign(this, new ARIAGlobalStatesAndProperties()); // Assigns ARIA global states and properties to the element
+  }
+
+  // Attributes
+  /**
+   * The name of the select element
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: name
+   */
+  @attr
+  public name?: string;
+
+  /**
+   * The required boolean attribute of the select element
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: required
+   */
+  @attr({ mode: 'boolean' })
+  public required?: boolean;
+
+  /**
+   * The optional label of the select
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: label
+   */
+  @attr
+  public label?: string;
+
+  /**
+   * The autofocus attribute of the select
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: autofocus
+   */
+  @attr({ attribute: 'autofocus', mode: 'boolean' })
+  public autofocusAttr?: boolean;
+
+  /**
+   * The autocomplete attribute of the select
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: autocomplete
+   */
+  @attr
+  public autocomplete?: string;
+
+  /**
+   * The size of the select
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: control-size
+   */
+  @attr
+  public controlSize?: SelectSize;
+
+  /**
+   * The disabled state of the select
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: disabled
+   */
+  @attr({ mode: 'boolean' })
+  public disabled?: boolean;
+
+  // Observables
+  /**
+   * The options of the select
+   *
+   * @public
+   */
+  @observable
+  public options?: HTMLOptionElement[];
+
+  /**
+   * A boolean value that triggers an input event
+   *
+   * @public
+   */
+  @observable
+  public inputEventTrigger: boolean = false;
```

Comment:
I'm not sure I understand how this would work - why do we need a boolean?

Reply:
We don't any more. This was for the (wrong) previous implementation that didn't use the emit helper.